### PR TITLE
chore!(all): attempt to update beego to v2

### PR DESCRIPTION
**Pull request by roberChen**
# PLEASE REVIEW CAREFULLY!

BREAKING CHANGE: beego has update to v2, and **I don't guarantee the stability**

status:
- [x] build succeed
- [x] deprecate ```beego/v2/adapter``` and rewrite
- [x] test by running binary (seems no problem)

optional:
- ~~explain changes and why choose *this one* when similar api also exist~~ **don't use *adapter*, use newer api**
- write tests and test by ```go test``` [not started]

## why it's nessasary to deprecate beego/v2/adapter?
1. adapter is mainly used for keeping compatible with v1, and it's not the recommend v2 api (see [here](https://pkg.go.dev/github.com/beego/beego/v2/adapter))
2. adapter is mainly a wrapper of other modules in v2, which is not so effective
3. when using adapter with other packages at the same time, it's really easy to occur errors such as type errors. So we should only choose one of *adapter* or all other packages in v2 to use. Reason why not using adapter has explained above. (there has been some errors caused by mixed using of *adapter* and recommened v2 found and fixed)

## some risks of deprecating adapter
1. new api has some (significant) changes compared with old v1 one.
2. when migrate to v2, there might be some logical ignorance.
3. during the migration, new merge to master branch has to sync with this pr branch, which is a huge work and migh cause some errors

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
-	"encoding/gob"
 	"bytes"
+	"context"
+	"encoding/gob"
 	"errors"
-	"github.com/astaxie/beego"
-)
+	"time"
 
+	"github.com/astaxie/beego/logs"
```
Comment (roberChen) [inline cache/cache.go:10]: note: this import path has not been changed, will fix this later

File: go.mod [hunk 2]
```diff
@@ -5,21 +5,17 @@ go 1.12
 require (
 	github.com/PuerkitoBio/goquery v1.4.1
 	github.com/andybalholm/cascadia v1.2.0 // indirect
-	github.com/astaxie/beego v1.10.1
+	github.com/astaxie/beego v1.12.3
```
Comment (roberChen) [inline go.mod:8]: note: this will remove later

**Comment by roberChen** [comment]
note:
the v2.0.1 of beego has bugs. For ```github.com/mindoc-org/mindoc/commands.RegisterModel()```, this function calls ```github.com/beego/beego/v2/adapter/orm.RegisterModelWithPrefix(...)```, in v2.0.1, the function has a bug, which has been fixed in https://github.com/beego/beego/pull/4386/

**Comment by roberChen** [comment]
current build of new version seems failed to create new admin user and demo book when initializing database (when running ```mindoc install```), however, when created the database using master build and switch to this new version, logins are succeeded.

After investigation, using beego/v2 version cannot create user or book when install. The database remains empty after *trying to insert admin and demo book*. REASON STILL IN INVESTIGATION

**Comment by roberChen** [comment]
# about orm api change in v2

v2 version of beego remove the o.Rollback api for transaction operation.
typically, in v1, you can write code like this:
```go
                o := orm.NewOrm()
                if err := o.Operateion(); err != nil {
                        o.Rollback()
                        ...
                }
```
however, in v2, this is not available. beego will handles the transaction in new way using
cluster. the new code is like below:
```go
                o := orm.NewOrm()
                if err := o.DoTx(func(ctx context.Context, txOrm orm.TxOrmer) error{
                        err := o.Operations()
                        if err != nil {
                                return err
                        }
                }); err != nil {
                        ...
                }
```
when operation failed, it will automatically calls o.Rollback() for TxOrmer.
more details see https://beego.me/docs/mvc/model/transaction.md

File: models/BookResult.go [hunk 3]
```diff
@@ -373,7 +374,7 @@ func (m *BookResult) Converter(sessionId string) (ConvertBookResult, error) {
 		}
 		var buf bytes.Buffer
 
-		if err := beego.ExecuteViewPathTemplate(&buf, "document/export.tpl", viewPath, map[string]interface{}{"Model": m, "Lists": item, "BaseUrl": conf.BaseUrl}); err != nil {
+		if err := adapter.ExecuteViewPathTemplate(&buf, "document/export.tpl", viewPath, map[string]interface{}{"Model": m, "Lists": item, "BaseUrl": conf.BaseUrl}); err != nil {
```
Comment (roberChen) [inline models/BookResult.go:377]: need change!

**Review by gsw945 — APPROVED** [review]
运行注意：本地运行，使用`http://127.0.0.1:8181/`访问时，需要修改配置文件`conf/app.conf`中的`enablexsrf`为`false`，否则**POST**请求(eg: 项目创建、文章创建等)会报xsrf错误。
考虑到一直以来(特别是无IT技术的)用户无需修改配置文件即可运行，升级beego版本到v2时，如果默认需要修改配置文件的行为，会给用户带来不便。
所以，建议将配置文件修改`enablexsrf`默认值设置为`false`，**README.md**文档或**wiki**中添加`enablexsrf`配置相关的安全性说明。

**Comment by LawyZheng** [comment]
beego v2.0.1 still leaves some problems unfixed, such as the Cookie SameSite feature, according to https://github.com/beego/beego/issues/4367.
I'm not sure whether it will cause some problems in Mindoc, since many functions in Mindoc are based on Cookies.
Maybe it's better for us to wait until beego v2 becomes stable.
Or at least, merge this feature into the dev branch or Mindoc 3.0 branch, rather than the master branch.

**Comment by roberChen** [comment]
> beego v2.0.1 still leaves some problems unfixed, such as the Cookie SameSite feature, according to [beego/beego#4367](https://github.com/beego/beego/issues/4367).
> I'm not sure whether it will cause some problems in Mindoc, since many functions in Mindoc are based on Cookies.
> Maybe it's better for us to wait until beego v2 becomes stable.
> Or at least, merge this feature into the dev branch or Mindoc 3.0 branch, rather than the master branch.

Actially I'm not using v2.0.1, there has several bugs on that version. I use the latest commit of beego v2, and the bug you've mensioned has been fixed.

**Review by gsw945 — APPROVED** [review]

File: models/Member.go [hunk 4]
```diff
@@ -118,32 +118,38 @@ func (m *Member) TmpLogin(account string) (*Member, error) {
 
 //ldapLogin 通过LDAP登陆
 func (m *Member) ldapLogin(account string, password string) (*Member, error) {
-	if beego.AppConfig.DefaultBool("ldap_enable", false) == false {
+	if web.AppConfig.DefaultBool("ldap_enable", false) {
```
Comment (wangbin8909) [inline models/Member.go:121]: 这里把 == false 去掉了，是原来的有问题，还是修改的不对。

File: commands/migrate/migrate.go [hunk 5]
```diff
@@ -113,17 +114,17 @@ func RunMigration() {
 
 //导出数据库的表结构
 func ExportDatabaseTable() ([]string, error) {
-	db_adapter := beego.AppConfig.String("db_adapter")
-	db_database := beego.AppConfig.String("db_database")
+	dbadapter,_  := web.AppConfig.String("db_adapter")
+	dbdatabase,_ := web.AppConfig.String("db_database")
```
Comment (wangbin8909) [inline commands/migrate/migrate.go:118]: 逗号后面加个空格

**Comment by roberChen** [comment]
@LawyZheng we could release a minor version marked as stable version and then develop on master branch later. Once this pr is merged, later development should based on this because beego v2 is quite a big difference compared with v1, so parallel dev is impossible

**Review by wangbin8909 — CHANGES_REQUESTED** [review]

File: models/Member.go (hunk 4 above)
Comment (roberChen) [reply models/Member.go]: 是修改的不对。已修改

File: commands/migrate/migrate.go (hunk 5 above)
Comment (roberChen) [reply commands/migrate/migrate.go]: 已format

**Review by gsw945 — DISMISSED** [review]

**Comment by roberChen** [comment]
build failed. travis.ci are using dep which is failed.

**Comment by gsw945** [comment]
> build failed. travis.ci are using dep which is failed for missing Gopkg, see [here](https://travis-ci.com/github/mindoc-org/mindoc/builds/221865466#L233)
> 
> I think the travis.ci should at least update the mod manager. Also go version could update.

[`.travis.yml`](https://github.com/mindoc-org/mindoc/blob/master/.travis.yml) for [Travis CI](https://travis-ci.com/github/mindoc-org/mindoc) is Editting, please wait.

**Review by gsw945 — APPROVED** [review]

**Review by wangbin8909 — COMMENTED** [review]
ok

**Review by wangbin8909 — APPROVED** [review]
approve

**Comment by roberChen** [comment]
@gsw945 could you please check and merge?

**Review by LawyZheng — APPROVED** [review]

**Pull request merged**